src/app/visit/visit.cpp

```diff
@@ -0,0 +1,59 @@
+#include "visit.hpp"
+#include "../doctor/doctor.hpp"
+
+std::set<Visit *> Visit::visitExtent;
+
+Visit::Visit()
+{
+    visitExtent.insert(this);
+}
+
+Visit::~Visit()
+{
+    removeFromExtent(this);
+}
+
+std::shared_ptr<Visit> Visit::createVisit(std::shared_ptr<Doctor> doc)
+{
+    auto visit = std::shared_ptr<Visit>(new Visit());
+    doc->addAssociation(visit);
+    return visit;
+}
+
+std::set<Visit *> Visit::getExtent()
+{
+    return visitExtent;
+}
+
+void Visit::removeFromExtent(Visit *visit)
+{
+    auto it = visitExtent.find(visit);
+    if (it != visitExtent.end())
+    {
+        visitExtent.erase(it);
+    }
+}
+
+void Visit::setAssociation(std::shared_ptr<Doctor> doc)
+{
+    if (!docAssociation_)
+    {
+        docAssociation_ = {doc};
+        docAssociation_->addAssociation(shared_from_this());
+    }
+}
+
+std::shared_ptr<Doctor> Visit::getDocAssociation()
+{
+    return docAssociation_;
+}
+
+void Visit::setVisitInfo(std::string info)
```

Comment:
I think, that in final version it should set VisitInfo struct (please read the rationale in hpp file for visitInfo_ field)

Of course the function could set a string, but it would create problems with string validation (does it contain all necessary info, with correct order etc. ?).